crates/aptos/src/move_tool/bytecode.rs

```diff
@@ -0,0 +1,306 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::common::{
+    types::{CliCommand, CliError, CliTypedResult, PromptOptions},
+    utils::{
+        check_if_file_exists, create_dir_if_not_exist, read_dir_files, read_from_file,
+        write_to_user_only_file,
+    },
+};
+use anyhow::Context;
+use async_trait::async_trait;
+use clap::Parser;
+use itertools::Itertools;
+use move_binary_format::{
+    binary_views::BinaryIndexedView, file_format::CompiledScript, CompiledModule,
+};
+use move_bytecode_source_map::{mapping::SourceMapping, utils::source_map_from_file};
+use move_command_line_common::files::{
+    MOVE_COMPILED_EXTENSION, MOVE_EXTENSION, SOURCE_MAP_EXTENSION,
+};
+use move_coverage::coverage_map::CoverageMap;
+use move_disassembler::disassembler::{Disassembler, DisassemblerOptions};
+use move_ir_types::location::Spanned;
+use std::{
+    fs,
+    path::{Path, PathBuf},
+    process::Command,
+};
+
+const REVELA_EXE_ENV: &str = "REVELA_EXE";
+#[cfg(target_os = "windows")]
+const REVELA_EXE: &str = "revela.exe";
+#[cfg(not(target_os = "windows"))]
+const REVELA_EXE: &str = "revela";
+
+const DISASSEMBLER_EXTENSION: &str = "mv.asm";
+const DECOMPILER_EXTENSION: &str = "mv.move";
+
+/// Disassemble the Move bytecode pointed to in the textual representation
+/// of Move bytecode.
+///
+/// For example, if you want to disassemble an on-chain package `PackName` at account `0x42`:
+/// 1. Download the package with `aptos move download --account 0x42 --package PackName --bytecode`
+/// 2. Disassemble the package bytecode with `aptos disassemble --package-path PackName/bytecode_modules`
+#[derive(Debug, Parser)]
+pub struct Disassemble {
+    #[clap(flatten)]
+    pub command: BytecodeCommand,
+}
+
+/// Decompile the Move bytecode pointed to into Move source code.
+///
+/// For example, if you want to decompile an on-chain package `PackName` at account `0x42`:
+/// 1. Download the package with `aptos move download --account 0x42 --package PackName --bytecode`
+/// 2. Decompile the package bytecode with `aptos decompile --package-path PackName/bytecode_modules`
+#[derive(Debug, Parser)]
+pub struct Decompile {
+    #[clap(flatten)]
+    pub command: BytecodeCommand,
+}
+
+#[derive(Debug, Clone, Copy, Default)]
+enum BytecodeCommandType {
+    #[default]
+    Disassemble,
```

Comment:
Interesting, we don't want Decompile as default?

I suspect most can't read assembly (or is it really expensive / less accurate)

Reply:
This is just there for technical reasons, not some actual default. Added a comment to clarify.